example_10/controllers/gpio_controller.cpp

```diff
@@ -0,0 +1,162 @@
+// Copyright 2023 ros2_control Development Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "ros2_control_demo_example_10/gpio_controller.hpp"
+
+#include <string>
+
+namespace ros2_control_demo_example_10
+{
+controller_interface::CallbackReturn GPIOController::on_init()
+{
+  try
+  {
+    auto_declare<std::vector<std::string>>("inputs", std::vector<std::string>());
+    auto_declare<std::vector<std::string>>("outputs", std::vector<std::string>());
+  }
+  catch (const std::exception & e)
+  {
+    fprintf(stderr, "Exception thrown during init stage with message: %s \n", e.what());
+    return controller_interface::CallbackReturn::ERROR;
+  }
+  return controller_interface::CallbackReturn::SUCCESS;
+}
+
+controller_interface::InterfaceConfiguration GPIOController::command_interface_configuration() const
+{
+  controller_interface::InterfaceConfiguration config;
+  config.type = controller_interface::interface_configuration_type::INDIVIDUAL;
+
+  for (auto command_interfaces_name : outputs_)
+  {
+    config.names.emplace_back(command_interfaces_name);
+  }
```

Comment:
```suggestion
  config.names = outputs_;
```

I'm not sure, if we can do as I suggested

Reply:
I'll try it